### PR TITLE
#47: Make release timeout configurable 

### LINKED ISSUE #47: Make timeout configurable
# Background

- Since, adding the Pandas 2 support to the script-languages-release repo, we have 300+ more tests which increase the CI execution time to above 2 hours
- For uploading the container as release artifact to Github, we currently use the Github Access Token of the Workflow run
- The Github Access Token of the Workflow run is only valid until the workflow ends
- If the timeout is causing the Workflow run to end, the AWS Code Build run can't upload the container as release artifact
- By making this timeout configurable in a config file, we can avoid changing the code of this repo, if further increases should be necessary.

https://github.com/exasol/script-languages-container-ci-setup/blob/03505cd6b5ed52bc85a834bb549c9138f2b68e5a/exasol_script_languages_container_ci_setup/lib/aws_access.py#L125

### PR DESCRIPTION
**Pull request by tkilias**
Fixes #47 

- Using Pydantic for modeling the Config
- Sorting the test into unit_tests and integration_tests
- Add parameter timeout_time_in_seconds to aws_access.start_codebuild
- Add boto mock tests for AWSAccess

### CODE REVIEW

File: test/unit_tests/test_config_data_model.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+from pathlib import Path
+
+import pytest
+
+from exasol_script_languages_container_ci_setup.lib.config.config_data_model import Config, Build, Ignore, Release
+from exasol_script_languages_container_ci_setup.lib.config.pydantic_model_generator import generate_config_data_model, \
+    CONFIG_DATA_MODEL_FILE_NAME
+
+
+def test():
```
Comment (Nicoretti) [inline test/unit_tests/test_config_data_model.py:11]: test as name?
No assert?

**Review by Nicoretti — APPROVED** [review]

File: test/unit_tests/test_pydantic_model_generator.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                    "a/b/c",
+                    "e/f/g"
+                ]
+            ),
+            base_branch=""
+        ),
+        release=module.Release(
+            timeout_in_minutes=1
+        )
+    )
```
[the remark this answers is not in the document]
Comment (tkilias) [reply test/unit_tests/test_pydantic_model_generator.py:37]: Not sure what you exactly mean

File: test/unit_tests/test_config_data_model.py (hunk 1 above)
Comment (tkilias) [reply test/unit_tests/test_config_data_model.py:11]: good point, will improve

File: exasol_script_languages_container_ci_setup/lib/config/pydantic_model_generator.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+from datamodel_code_generator import generate, InputFileType
+
+from exasol_script_languages_container_ci_setup.lib.render_template import render_template
+
+logger = logging.getLogger(__name__)
+
+CONFIG_DATA_MODEL_FILE_NAME = "config_data_model.py"
+
+
+def get_config_data_model_pydantic_model_default_output_file() -> Path:
```
[the remark this answers is not in the document]
Comment (tkilias) [reply exasol_script_languages_container_ci_setup/lib/config/pydantic_model_generator.py:15]: you are right, the name isn't helpful, will think of a better one

File: test/unit_tests/test_config_data_model.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        release=Release(
+            timeout_in_minutes=1
+        )
+    )
+    return config
+
+
+def test_serialization(expected_config, expected_config_json):
+    actual_json = expected_config.json(indent=4)
+    print(actual_json)
```
Comment (Nicoretti) [inline test/unit_tests/test_config_data_model.py:29]: debugging leftover?

File: test/unit_tests/conftest.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
+from inspect import cleandoc
+
+import pytest
+
+
+@pytest.fixture
+def expected_config_json() -> str:
```
Comment (Nicoretti) [inline test/unit_tests/conftest.py:7]:
```suggestion
def expected_json_config() -> str:
```

File: test/unit_tests/test_config_data_model.py (hunk 4 above)
Comment (tkilias) [reply test/unit_tests/test_config_data_model.py:29]:
```suggestion
```

File: test/unit_tests/test_config_data_model.py [hunk 6]
```diff
@@ -26,7 +26,6 @@ def expected_config() -> Config:
 
 def test_serialization(expected_config, expected_config_json):
     actual_json = expected_config.json(indent=4)
-    print(actual_json)
     assert actual_json == expected_config_json
```
Comment (Nicoretti) [inline test/unit_tests/test_config_data_model.py]: I think you missed to update the test with the new fixture name

**Review by Nicoretti — APPROVED** [review]

**Pull request merged**